pyproject.toml

```diff
@@ -1,3 +1,150 @@
 [tool.black]
 line-length = 100
 include = '\.pyi?$'
+
+[tool.poetry]
+name = "pennylane"
+version = "0.35.0-dev"
+description = "PennyLane is a Python quantum machine learning library by Xanadu Inc."
+authors = ["PennyLaneAI <info@pennylane.ai>"]
+license = "Apache-2.0"
+readme = "README.md"
+homepage = "https://pennylane.ai/"
+repository = "https://github.com/PennyLaneAI/pennylane"
+
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry.plugins."pennylane.plugins"]
+"default.qubit" = "pennylane.devices:DefaultQubit"
+"default.qubit.legacy" = "pennylane.devices:DefaultQubitLegacy"
+"default.gaussian" = "pennylane.devices:DefaultGaussian"
+"default.clifford" = "pennylane.devices:DefaultClifford"
+"default.qubit.tf" = "pennylane.devices.default_qubit_tf:DefaultQubitTF"
+"default.qubit.torch" = "pennylane.devices.default_qubit_torch:DefaultQubitTorch"
+"default.qubit.autograd" = "pennylane.devices.default_qubit_autograd:DefaultQubitAutograd"
+"default.qubit.jax" = "pennylane.devices.default_qubit_jax:DefaultQubitJax"
+"default.mixed" = "pennylane.devices.default_mixed:DefaultMixed"
+"null.qubit" = "pennylane.devices.null_qubit:NullQubit"
+"default.qutrit" = "pennylane.devices.default_qutrit:DefaultQutrit"
+
```

Comment:
Can all poetry-specific sections here and below instead conform with the standardized PEP (e.g. 517/621/etc) ways of specifying these sections? I'd rather we didn't have poetry-specifics in something that is used to send the ecosystem out into the wild, but is only used to help manage the CI and inter-related deps therein.

Adding anything poetry specific here that becomes ingrained cannot then be used with pip/conda/etc, as it isn't standardized by Python.

Reply:
I tried to change it to use non-poetry stuff locally, but poetry can only interpret its own headers. I imagine that they align for the most part, but if we intend on adding non-poetry configuration (eg. `[project]` instead of `[tool.poetry]`, and `[project.entry-points."pennylane.plugins"]` instead of `[tool.poetry.plugins."pennylane.plugins"]`), there will just have to be a lot of duplication. Those cases are just `s/tool.poetry/project`, but that won't always be the case.

[The README for their build-system](https://github.com/python-poetry/poetry-core) makes me think that we will in fact require people to have `poetry-core` installed in order to `pip install pennylane`, but that should only be from source.

Reply:
Thanks @timmysilv 
I think we should discuss this more next week with @trbromley  and @Alex-Preciado and see what makes the most sense here.